<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1350</width>
    <height>877</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="m_centralWidget"/>
  <widget class="QStatusBar" name="_statusBar"/>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1350</width>
     <height>25</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFILE">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOpenMesh"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuMisc">
    <property name="title">
     <string>Materials</string>
    </property>
    <addaction name="actionReload_Shaders"/>
    <addaction name="actionOpen_Material_Editor"/>
   </widget>
   <widget class="QMenu" name="menuKeymapping">
    <property name="title">
     <string>Keymapping</string>
    </property>
    <addaction name="actionReload_configuration"/>
    <addaction name="separator"/>
    <addaction name="actionLoad_configuration_file"/>
   </widget>
   <addaction name="menuFILE"/>
   <addaction name="menuMisc"/>
   <addaction name="menuKeymapping"/>
  </widget>
  <widget class="QDockWidget" name="dockWidget">
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents">
    <layout class="QGridLayout" name="gridLayout">
     <item row="0" column="0">
      <widget class="QWidget" name="widget" native="true">
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_4">
            <item>
             <widget class="QTreeView" name="m_entitiesTreeView">
              <property name="minimumSize">
               <size>
                <width>0</width>
                <height>150</height>
               </size>
              </property>
              <property name="editTriggers">
               <set>QAbstractItemView::NoEditTriggers</set>
              </property>
              <property name="textElideMode">
               <enum>Qt::ElideRight</enum>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="selectionLayout">
              <item>
               <widget class="QLabel" name="label_selected">
                <property name="text">
                 <string>Selected Object :</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="m_selectedItemName">
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QGridLayout" name="gridLayout_4">
              <item row="0" column="2">
               <widget class="QPushButton" name="m_editRenderObjectButton">
                <property name="enabled">
                 <bool>false</bool>
                </property>
                <property name="text">
                 <string>Edit</string>
                </property>
               </widget>
              </item>
              <item row="0" column="1">
               <widget class="QPushButton" name="m_toggleRenderObjectButton">
                <property name="text">
                 <string>Toggle Visible</string>
                </property>
               </widget>
              </item>
              <item row="0" column="0">
               <widget class="QPushButton" name="m_removeEntityButton">
                <property name="text">
                 <string>Remove</string>
                </property>
               </widget>
              </item>
              <item row="1" column="0">
               <widget class="QPushButton" name="m_exportMeshButton">
                <property name="text">
                 <string>Export Mesh</string>
                </property>
               </widget>
              </item>
              <item row="1" column="1">
               <widget class="QPushButton" name="m_clearSceneButton">
                <property name="text">
                 <string>Clear Scene</string>
                </property>
               </widget>
              </item>
              <item row="1" column="2">
               <widget class="QPushButton" name="m_fitCameraButton">
                <property name="text">
                 <string>Fit Camera</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="shaderLayout">
            <item>
             <widget class="QLabel" name="label">
              <property name="text">
               <string>Current shader</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QComboBox" name="m_currentShaderBox">
              <item>
               <property name="text">
                <string/>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Plain</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>BlinnPhong</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>LinesGeom</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Lines</string>
               </property>
              </item>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QTabWidget" name="toolBox">
            <property name="minimumSize">
             <size>
              <width>250</width>
              <height>0</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>16777215</width>
              <height>500</height>
             </size>
            </property>
            <property name="tabPosition">
             <enum>QTabWidget::West</enum>
            </property>
            <property name="currentIndex">
             <number>0</number>
            </property>
            <widget class="QWidget" name="tab_stats">
             <attribute name="title">
              <string>Stats</string>
             </attribute>
             <layout class="QVBoxLayout" name="verticalLayout_10">
              <item>
               <layout class="QFormLayout" name="formLayout_6">
                <item row="0" column="0">
                 <widget class="QLabel" name="label_23">
                  <property name="text">
                   <string>Frames averaging over count</string>
                  </property>
                 </widget>
                </item>
                <item row="0" column="1">
                 <widget class="QSpinBox" name="m_avgFramesCount">
                  <property name="minimum">
                   <number>1</number>
                  </property>
                  <property name="maximum">
                   <number>1000</number>
                  </property>
                  <property name="singleStep">
                   <number>10</number>
                  </property>
                  <property name="value">
                   <number>100</number>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <widget class="QLabel" name="m_frameA2BLabel">
                <property name="text">
                 <string>Frames #a to #b stats :</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="m_labelCount">
                <property name="text">
                 <string>Rendering #f faces and #v vertices</string>
                </property>
               </widget>
              </item>
              <item>
               <layout class="QGridLayout" name="gridLayout_6">
                <item row="2" column="0">
                 <widget class="QLabel" name="label_26">
                  <property name="text">
                   <string>Render</string>
                  </property>
                 </widget>
                </item>
                <item row="0" column="0">
                 <widget class="QLabel" name="label_25">
                  <property name="text">
                   <string/>
                  </property>
                 </widget>
                </item>
                <item row="3" column="0">
                 <widget class="QLabel" name="label_27">
                  <property name="text">
                   <string>Tasks</string>
                  </property>
                 </widget>
                </item>
                <item row="0" column="1">
                 <widget class="QLabel" name="label_29">
                  <property name="text">
                   <string>Average time (Âµs)</string>
                  </property>
                 </widget>
                </item>
                <item row="4" column="0">
                 <widget class="QLabel" name="label_28">
                  <property name="text">
                   <string>Frame</string>
                  </property>
                 </widget>
                </item>
                <item row="0" column="2">
                 <widget class="QLabel" name="label_30">
                  <property name="text">
                   <string>Updates / second</string>
                  </property>
                 </widget>
                </item>
                <item row="2" column="2">
                 <widget class="QLabel" name="m_renderUpdates">
                  <property name="text">
                   <string>render</string>
                  </property>
                  <property name="alignment">
                   <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                  </property>
                 </widget>
                </item>
                <item row="3" column="2">
                 <widget class="QLabel" name="m_tasksUpdates">
                  <property name="text">
                   <string>tasks</string>
                  </property>
                  <property name="alignment">
                   <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                  </property>
                 </widget>
                </item>
                <item row="4" column="2">
                 <widget class="QLabel" name="m_frameUpdates">
                  <property name="text">
                   <string>frame</string>
                  </property>
                  <property name="alignment">
                   <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                  </property>
                 </widget>
                </item>
                <item row="2" column="1">
                 <widget class="QLabel" name="m_renderTime">
                  <property name="text">
                   <string>render</string>
                  </property>
                  <property name="alignment">
                   <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                  </property>
                 </widget>
                </item>
                <item row="3" column="1">
                 <widget class="QLabel" name="m_tasksTime">
                  <property name="text">
                   <string>tasks</string>
                  </property>
                  <property name="alignment">
                   <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                  </property>
                 </widget>
                </item>
                <item row="4" column="1">
                 <widget class="QLabel" name="m_frameTime">
                  <property name="text">
                   <string>frame</string>
                  </property>
                  <property name="alignment">
                   <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                  </property>
                 </widget>
                </item>
                <item row="1" column="0">
                 <widget class="QLabel" name="label_4">
                  <property name="text">
                   <string>Events</string>
                  </property>
                 </widget>
                </item>
                <item row="1" column="1">
                 <widget class="QLabel" name="m_eventsTime">
                  <property name="text">
                   <string>events</string>
                  </property>
                  <property name="alignment">
                   <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                  </property>
                  <property name="events" stdset="0">
                   <string>render</string>
                  </property>
                 </widget>
                </item>
                <item row="1" column="2">
                 <widget class="QLabel" name="m_eventsUpdates">
                  <property name="text">
                   <string>events</string>
                  </property>
                  <property name="alignment">
                   <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <layout class="QFormLayout" name="formLayout_7">
                <item row="1" column="0">
                 <widget class="QLabel" name="label_24">
                  <property name="text">
                   <string>Average framerate :</string>
                  </property>
                 </widget>
                </item>
                <item row="1" column="1">
                 <widget class="QLabel" name="m_avgFramerate">
                  <property name="text">
                   <string>fps</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout">
                <property name="bottomMargin">
                 <number>0</number>
                </property>
                <item>
                 <widget class="QCheckBox" name="m_printGraph">
                  <property name="text">
                   <string>Print Graph</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QCheckBox" name="m_printTimings">
                  <property name="text">
                   <string>Print Timings</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <spacer name="verticalSpacer">
                <property name="orientation">
                 <enum>Qt::Vertical</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>20</width>
                  <height>40</height>
                 </size>
                </property>
               </spacer>
              </item>
             </layout>
            </widget>
            <widget class="Ra::Gui::TransformEditorWidget" name="tab_edition">
             <attribute name="title">
              <string>Edition</string>
             </attribute>
            </widget>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QGridLayout" name="gridLayout_3" rowstretch="0,0,0,0,0" columnstretch="1,0">
          <item row="3" column="0">
           <widget class="QLabel" name="label_6">
            <property name="text">
             <string>Enable Debug Drawing</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_2">
            <property name="text">
             <string>Displayed Texture</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QCheckBox" name="m_enablePostProcess">
            <property name="enabled">
             <bool>true</bool>
            </property>
            <property name="text">
             <string/>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
            <property name="tristate">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item row="4" column="1">
           <widget class="QCheckBox" name="m_realFrameRate">
            <property name="enabled">
             <bool>true</bool>
            </property>
            <property name="text">
             <string/>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
            <property name="tristate">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QCheckBox" name="m_enableDebugDraw">
            <property name="enabled">
             <bool>true</bool>
            </property>
            <property name="text">
             <string/>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
            <property name="tristate">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_5">
            <property name="text">
             <string>Enable Post-Process</string>
            </property>
           </widget>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="label_7">
            <property name="text">
             <string>Variable Frame Rate</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QComboBox" name="m_displayedTextureCombo"/>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="label_8">
            <property name="text">
             <string>Current Renderer</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QComboBox" name="m_currentRendererCombo"/>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>LeftToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="separator"/>
   <addaction name="actionToggle_Local_Global"/>
   <addaction name="actionGizmoOff"/>
   <addaction name="actionGizmoTranslate"/>
   <addaction name="actionGizmoRotate"/>
   <addaction name="separator"/>
   <addaction name="actionRecord_Frames"/>
   <addaction name="separator"/>
  </widget>
  <action name="actionOpenMesh">
   <property name="text">
    <string>Open...</string>
   </property>
   <property name="toolTip">
    <string>Open...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionReload_Shaders">
   <property name="text">
    <string>Reload Shaders</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+R</string>
   </property>
  </action>
  <action name="actionGizmoTranslate">
   <property name="checkable">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset>
     <normalon>:/Assets/Images/translate.png</normalon>
    </iconset>
   </property>
   <property name="text">
    <string>Translate</string>
   </property>
   <property name="toolTip">
    <string>Translate selected object</string>
   </property>
  </action>
  <action name="actionGizmoRotate">
   <property name="checkable">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset>
     <normalon>:/Assets/Images/rotate.png</normalon>
    </iconset>
   </property>
   <property name="text">
    <string>Rotate</string>
   </property>
   <property name="toolTip">
    <string>Rotate selected object</string>
   </property>
  </action>
  <action name="actionToggle_Local_Global">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset>
     <normalon>:/Assets/Images/gizmo.png</normalon>
    </iconset>
   </property>
   <property name="text">
    <string>Toggle Local/Global</string>
   </property>
   <property name="toolTip">
    <string>Changes the transform edition mode</string>
   </property>
  </action>
  <action name="actionGizmoOff">
   <property name="checkable">
    <bool>false</bool>
   </property>
   <property name="checked">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset>
     <normalon>:/Assets/Images/select.png</normalon>
    </iconset>
   </property>
   <property name="text">
    <string>No Gizmos</string>
   </property>
   <property name="toolTip">
    <string>Disable transform gizmo.</string>
   </property>
  </action>
  <action name="actionOpen_Material_Editor">
   <property name="text">
    <string>Open Material Editor</string>
   </property>
   <property name="shortcut">
    <string>Alt+M</string>
   </property>
  </action>
  <action name="actionRecord_Frames">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../../../../icons.qrc">
     <normaloff>:/Assets/Images/rec.png</normaloff>
     <normalon>:/Assets/Images/rec_on.png</normalon>:/Assets/Images/rec.png</iconset>
   </property>
   <property name="text">
    <string>Record Frames</string>
   </property>
  </action>
  <action name="actionReload_configuration">
   <property name="text">
    <string>Reload configuration</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+R</string>
   </property>
  </action>
  <action name="actionLoad_configuration_file">
   <property name="text">
    <string>Load configuration file...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+O</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>Ra::Gui::TransformEditorWidget</class>
   <extends>QWidget</extends>
   <header>Gui/TransformEditorWidget.hpp</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../../../../icons.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionExit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>518</x>
     <y>373</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
